### DISCUSSION TITLE
Next.js 14 apps on FreeBSD: @next/swc-freebsd-x64 package is missing/not downloaded

### DISCUSSION THREAD

**Discussion by wekinglypigs** [vercel/next.js#59634] [Help]
### Summary

When I make a minimal app on FreeBSD with (for example) `npx create-next-app -e reproduction-template` and try to `npm run dev` or `npm run build` it fails to download swc-freebsd-x64 package with status 404.  Here is some output including the error message:

```
% npm run dev

> dev
> next dev

   - Next.js 14.0.5-canary.12
   - Local:        http://localhost:3000

   Downloading swc package @next/swc-freebsd-x64...
 -¨¯ Failed to download swc package from https://registry.npmjs.org/@next/swc-freebsd-x64/-/swc-freebsd-x64-14.0.5-canary.12.tgz
Error: request failed with status 404
```

If I run `npm show @next/swc-freebsd-x64 versions` there are many versions available (through 13.2.5) but no later (14.0) versions.  How to fix this, will current versions be provided?

### Additional information

```js
OS: FreeBSD 14.0-RELEASE-p3 amd64
node: v20.8.1
npm: 10.2.0

% cat package.json
{
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start"
  },
  "dependencies": {
    "next": "canary",
    "react": "^18.2.0",
    "react-dom": "^18.2.0"
  },
  "devDependencies": {
    "@types/node": "20.4.5",
    "@types/react": "18.2.18",
    "typescript": "5.1.3"
  }
}
```

**Comment by vonpoland** [comment]
Same here. Seems @next/swc-freebsd-x64 14.0.4 is missing in npm repo...
https://www.npmjs.com/package/@next/swc-freebsd-x64?activeTab=versions

```
> next build

 ⚠ Disabling SWC Minifer will not be an option in the next major version. Please report any issues you may be experiencing to https://github.com/vercel/next.js/issues
   ▲ Next.js 14.0.4

   Downloading swc package @next/swc-freebsd-x64...
 ⨯ Failed to download swc package from https://registry.npmjs.org/@next/swc-freebsd-x64/-/swc-freebsd-x64-14.0.4.tgz
unhandledRejection Error: request failed with status 404
```

seems last time this issue fixed it https://github.com/vercel/next.js/issues/36639 @Brooooooklyn can you take a look again ?

**Comment by tinola** [comment]
Same here.
I can't build [ztnet](https://github.com/sinamics/ztnet) application.

My system: FreeBSD 14.0-RELEASE-p3

Errorlog:
```

> ztnet@0.1.0 build
> next build

   Skipping linting
 ✓ Checking validity of types
   ▲ Next.js 14.0.3
   - Environments: .env
   - Experiments (use at your own risk):
     · instrumentationHook

   Downloading swc package @next/swc-freebsd-x64...
 ⨯ Failed to download swc package from https://registry.npmjs.org/@next/swc-freebsd-x64/-/swc-freebsd-x64-14.0.3.tgz
unhandledRejection Error: request failed with status 404

```

Thanks for any help to get support in FreeBSD 14

**Comment by wagnerflo** [comment]
Encountering the same thing. Is FreeBSD not supported any more? Can someone take a look? @Brooooooklyn?

**Comment by agsimfzi** [comment]
Same issue here:
`> next dev                          
                                     
  ▲ Next.js 14.2.4                   
  - Local:        http://localhost:3000                                    
  - Environments: .env

 ✓ Starting...    
   Downloading swc package @next/swc-freebsd-x64...                        
 ⨯ Failed to download swc package from https://registry.npmjs.org/@next/swc-freebsd-x64/-/swc-freebsd-x64-14.2.4.tgz                                   
Error: request failed with status 404`

**Comment by fearedbliss** [comment]
Same here on `FreeBSD 14.1-STABLE stable/14-n268370-b37a6d41a046 GENERIC amd64`.

**Comment by chriswells0** [comment]
I was able to work around this on FreeBSD 13.3 by following the instructions here:

https://nextjs.org/docs/messages/failed-loading-swc

All it required was to create _.babelrc_:

```
{
  "presets": ["next/babel"]
}
```

It does sound like compilation will be slower, but that's better than being blocked.

**Comment by mikl** [comment]
Fallback to Babel is apparently not an option with Next.js v15, and installing `@next/swc-freebsd-x64` does not help. I now get segfaults when trying to build my app:

```
$ pnpm run build

> myapp@1.7.8 build /srv/www/sites/myapp.test
> NEXT_TELEMETRY_DISABLED=1 next build

   ▲ Next.js 15.2.4
   - Experiments (use with caution):
     · clientTraceMetadata

   Creating an optimized production build ...
Segmentation fault
```
